js/src/logger.ts

```diff
@@ -1367,6 +1491,19 @@ class BackgroundLogger {
       }
     }
 
+    const attachments = this.attachments;
+    this.attachments = [];
+    // TODO: Error handling.
+    const uploadResponses = await Promise.all(
```

Comment:
Would it make sense to do this concurrently with logging, rather than after?

Reply:
Also note that `Promise.all` can abort the entire set of promises if any of them fails. Maybe in this case we want to use `Promise.allSettled` to try resolving all of them?

Reply:
Done